core/src/test/java/fr/sncf/osrd/envelope_sim_infra/EnvelopeTrainPathTest.java

```diff
@@ -0,0 +1,144 @@
+package fr.sncf.osrd.envelope_sim_infra;
+
+import static fr.sncf.osrd.Helpers.getSmallInfra;
+import static fr.sncf.osrd.api.pathfinding.PathfindingUtils.makePath;
+import static fr.sncf.osrd.envelope_sim_infra.EnvelopeTrainPath.from;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import com.google.common.collect.ImmutableRangeMap;
+import com.google.common.collect.Range;
+import com.google.common.collect.TreeRangeMap;
+import fr.sncf.osrd.api.FullInfra;
+import fr.sncf.osrd.envelope_sim.electrification.Electrification;
+import fr.sncf.osrd.envelope_sim.electrification.Electrified;
+import fr.sncf.osrd.envelope_sim.electrification.Neutral;
+import fr.sncf.osrd.external_generated_inputs.ElectricalProfileMapping;
+import fr.sncf.osrd.railjson.schema.external_generated_inputs.RJSElectricalProfileSet;
+import fr.sncf.osrd.railjson.schema.infra.trackranges.RJSTrackRange;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import java.util.List;
+import java.util.Map;
+
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public class EnvelopeTrainPathTest {
+
+    private FullInfra infra;
+
+    @BeforeAll
+    public void setUp() {
+        infra = getSmallInfra();
+    }
+
+    @Test
+    public void envelopeFromPathTestElectrificationMap() {
+        var path = makePath(infra.blockInfra(), infra.rawInfra(), 0);
```

Comment:
Because we reimplement lots of projection logic, we need tests with paths that are more complex than that (several blocks, offsets, blocks that don't start at offset 0 on the track)

If we figure out how to reuse the path iteration functions, this wouldn't be as necessary

Reply:
done :)